src/Common/cmake/secure_dependencies.cmake

```diff
@@ -0,0 +1,73 @@
+#===============================================================================
+# Configure the flags used to ensure that required security settings are enabled
+macro(configure_security_flags)
+    # Enable Control Flow Guard
+    if (WIN32)
+        add_link_options("LINKER:/guard:cf")
+    endif()
+
+    # Enable Compiler supported Spectre mitigations
+    if (NOT APPLE)
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mspeculative-load-hardening -mretpoline")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mspeculative-load-hardening -mretpoline")
+    endif()
+endmacro()
+
+#===============================================================================
+# Set MSVC static runtime library policy
+# NOTE: Since this sets a cmake policy for ensuring Windows compilation uses static runtime,
+# this policy MUST be set before any projects are declared for it to take effect.
+macro(set_msvc_static_runtime_policy)
+    # Enforce use of static runtime (avoids target machine needing msvcrt installed).
+    # Must set policy CMP0091 before any projects are declared (see https://cmake.org/cmake/help/latest/variable/CMAKE_MSVC_RUNTIME_LIBRARY.html#cmake-msvc-runtime-library)
+    cmake_policy(SET CMP0091 NEW)
+    set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+endmacro()
+
+#===============================================================================
+# Locate Spectre mitigated static C runtime on Windows. Sets the `SPECTRE_LIBS`
+# variable for use in later calls to `target_link_libraries`.
+# NOTE: This sets a cmake policy for ensuring Windows compilation uses static runtime.
+# this policy MUST be set before any projects are declared for it to take effect.
+macro(locate_win32_spectre_static_runtime)
+    if (WIN32)
+        # Locate the vswhere application, which will provide paths to any installed Visual Studio instances.
+        # By invoking it with "-find **/lib/spectre/x64" we will find any Spectre mitigated libaries that
+        # have been installed.
+        find_program(_vswhere_tool
+            NAMES vswhere
+            PATHS "$ENV{ProgramFiles\(x86\)}/Microsoft Visual Studio/Installer")
+        if (NOT ${vswhere})
+            message(FATAL_ERROR "Could not locate vswhere.exe - unable to source vc redistributable")
```

Comment:
(nit)
If the users try to build locally and have no expected version of MS VC++ installed (e.g. our ReadMe, Prerequisites section, gets out-of-date and recommends to install the VS version older than needed), then the message "unable to source vc redistributable" may confuse them. Consider more detailed, more actionable message.

Reply:
Disregard. This is the old/existing code.